rest-api-model-server/src/main/kotlin/org/modelix/sample/restapimodelserver/UpdateSocket.kt

```diff
@@ -70,12 +64,16 @@ class UpdateSocket(private val repo: ReplicatedRepository, private val mapper: O
         val area = PArea(repo.branch)
         val node = PNodeAdapter(nodeId, repo.branch)
         area.executeRead {
-            when (node.concept) {
-                is RoomConcept -> broadcast(ChangeNotification(WhatChanged.ROOM, Room(node).toJson()))
-                is RoomsConcept -> broadcast(ChangeNotification(WhatChanged.ROOM_LIST, Rooms(node).toJson()))
-                is LectureConcept -> broadcast(ChangeNotification(WhatChanged.LECTURE, Lecture(node).toJson()))
-                is CoursesConcept -> broadcast(ChangeNotification(WhatChanged.LECTURE_LIST, Courses(node).toJson()))
-                else -> logger.warn("Could not handle change")
+            try {
+                when (node.typed()) {
+                    is N_Room -> broadcast(ChangeNotification(WhatChanged.ROOM, node.typed<N_Room>().toJson()))
+                    is N_Rooms -> broadcast(ChangeNotification(WhatChanged.ROOM_LIST, node.typed<N_Rooms>().rooms.toList().toJson()))
+                    is N_Lecture -> broadcast(ChangeNotification(WhatChanged.LECTURE, node.typed<N_Lecture>().toJson()))
+                    is N_Courses -> broadcast(ChangeNotification(WhatChanged.LECTURE_LIST, node.typed<N_Courses>().lectures.toList().toJson()))
+                    else -> logger.warn("Could not handle change")
+                }
+            } catch (e: RuntimeException){
+                logger.warn("Ignoring change due to invalid model: "+ e.message)
             }
```

Comment:
Why can this happen?

Reply:
When a concept is added in MPS, it has empty properties/childeren/references - even for required fields (1..n). Consequently, the API will raise a RuntimeException. This case is handled here. Also see the corresponding issue: https://github.com/modelix/modelix.core/issues/31

Reply:
Ah, could you put that issue into a comment? That would ensure the next reader knows why RuntimeException is caught here.